lib/ruby_smb/server/share/provider/virtual_disk/virtual_pathname.rb

```diff
@@ -0,0 +1,193 @@
+module RubySMB
+  class Server
+    module Share
+      module Provider
+        class VirtualDisk < Disk
+          # This object emulates Ruby's builtin Pathname object but uses a virtual file system instead of the real local
+          # one.
+          class VirtualPathname
+            SEPARATOR = File::SEPARATOR
+            # see: https://ruby-doc.org/stdlib-3.1.1/libdoc/pathname/rdoc/Pathname.html
+            STAT_METHODS = %i[
+              atime
+              birthtime
+              blockdev?
+              chardev?
+              ctime
+              directory?
+              executable?
+              file?
+              ftype
+              grpowned?
+              mtime
+              owned?
+              pipe?
+              readable?
+              setgid?
+              setuid?
+              size
+              socket?
+              sticky?
+              symlink?
+              world_readable?
+              world_writable?
+              writable?
+              zero?
+            ]
+            private_constant :STAT_METHODS
+
+            attr_accessor :virtual_disk
+
+            # @param [Hash] disk The mapping of paths to objects representing the virtual file system.
+            # @param [String] path The path of this entry.
+            # @param [Boolean] exist? Whether or not this entry represents an existing entry in the virtual file system.
+            # @param [File::Stat] stat An explicit stat that represents this object. A default VirtualStat will be
+            #   created unless specified.
+            def initialize(disk, path, **kwargs)
```

Comment:
Because that's invalid syntax 😭 

```
[1] pry(main)> def initialize(disk, path, exist?: true, stat: nil)
[1] pry(main)* end  
(eval):2: formal argument must be local variable
... initialize(disk, path, exist?: true, stat: nil)
...                        ^~~~~~~
(eval):2: identifier exist? is not valid to set
[1] pry(main)>
```

The options I considered were:
* (this one) keep the consistency of `exists?` which is a direct 1 to one mapping with the name of the attribute it sets
* use what you suggest and require the developer to know that all attributes that end in a `?` are set by keywords that do not have that suffix


This pattern because a bit more important when you look at the `VirtualStat` which allows all of it's attributes to be set from the keyword arguments. I was just worried it might confuse users if they had to know that the names were modified.

Reply:
Oh yes, you're right, it is not a valid syntax. I usually don't use '?' for this, like we do fo predicate method. I understand your point though and I'm fine with this.